File: Allegro5 Test/GreyFloor.cpp
#include "GreyFloor.h"


GreyFloor::GreyFloor(void)
{
}

void GreyFloor::Init(ALLEGRO_BITMAP *image = 0, float x = 0, float y = 0, int boundX = 100, int boundY = 100){
	Floor::Init(image, x, y, boundX, boundY);
	SetCollideType(GREY);
}

GreyFloor::~GreyFloor(void)
{
}

File: js/greyFloor.js
// GreyFloor class - floor that both characters can stand on

class GreyFloor extends Floor {
    constructor() {
        super();
        this.setCollideType(COLLIDE_TYPE.GREY);
    }
    
    init(texture, x, y, boundX, boundY) {
        super.init(texture, x, y, boundX, boundY);
        
        // Make the sprite appear grey
        if (this.sprite) {
            this.sprite.tint = 0x808080; // Grey tint
        }
    }
}